### PR TITLE
[FIX] Refactoring the status check for Task State in Project.task 

### PR DESCRIPTION
**Pull request by jananjoy**

### CODE REVIEW

**Comment by eilst** [comment]
squash your merges in 1 commit

File: project_event/models/project_task.py [hunk 1]
```diff
@@ -980,3 +980,10 @@ def get_confirmation_wizard(self, action):
             'target': 'new',
             'res_id': new_wizard.id,
         }
+
+    def check_task_state(self, task_state_in):
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: You didnt use the self

File: project_event/models/project_task.py [hunk 2]
```diff
@@ -980,3 +980,10 @@ def get_confirmation_wizard(self, action):
...
+        if task_state_in in \
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: simply return task_state_in in ...

Comment (jananjoy) [reply project_event/models/project_task.py]: @rimbendhaou ok

File: project_event/models/project_task.py (hunk 1 above)
Comment (jananjoy) [reply project_event/models/project_task.py]: @rimbendhaou  ok

**Review by rimbendhaou — APPROVED** [review]

**Review by rimbendhaou — CHANGES_REQUESTED** [review]
resolve conflicts

File: project_event/models/project_task.py [hunk 3]
```diff
@@ -804,7 +807,7 @@ def do_reservation(self):
 
     @api.multi
     def action_cancel(self):
-        if self.activity_task_type == 'task' and \
+        if self.is_type_task() and  \
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: extra space (PEP8)

File: project_event/tests/test_project_event_task.py [hunk 4]
```diff
@@ -527,3 +527,37 @@ def test_220_get_parent_project_id(self):
             self.task_2.get_parent_project_id(),
             self.project_2.id
         )
+
+    def test_230_is_type_task(self):
+        check_task_type_vals = {
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py]: you can simply use the self.task1 that is created in the common file

File: project_event/tests/test_project_event_task.py [hunk 5]
```diff
@@ -527,3 +527,37 @@ def test_220_get_parent_project_id(self):
...
+            'name': 'Sample Task for checking task type',
+            'activity_task_type': 'task',
+            'partner_id': self.project_1.partner_id.id,
+            'room_id': self.room_1.id,
+            'date_start': fields.Datetime.to_string(datetime.today()),
+            'date_end': fields.Datetime.to_string(datetime.today() +
+                                                  timedelta(hours=4)),
+        }
+        task = self.Tasks.create(check_task_type_vals)
+        self.assertEqual(task.name, 'Sample Task for checking task type')
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py]: no need to check this

File: project_event/tests/test_project_event_task.py [hunk 6]
```diff
@@ -527,3 +527,37 @@ def test_220_get_parent_project_id(self):
...
+        self.assertEqual(task.activity_task_type, 'task')
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py]: no need to check this

File: project_event/tests/test_project_event_task.py [hunk 7]
```diff
@@ -527,3 +527,37 @@ def test_220_get_parent_project_id(self):
...
+        check_task_type = task.is_type_task()
+        self.assertEqual(check_task_type, True)
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py]: simply : self.assertTrue(self.task1is_type_task())

File: project_event/tests/test_project_event_task.py [hunk 8]
```diff
@@ -527,3 +527,37 @@ def test_220_get_parent_project_id(self):
...
+
+    def test_240_check_task_state(self):
+        check_task_state_vals = {
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py]: same comment: use the task created in common

File: project_event/tests/test_project_event_task.py [hunk 9]
```diff
@@ -527,3 +527,37 @@ def test_220_get_parent_project_id(self):
...
+            'activity_task_type': 'task',
+            'partner_id': self.project_1.partner_id.id,
+            'room_id': self.room_1.id,
+            'task_state': 'option',
+            'date_start': fields.Datetime.to_string(datetime.today()),
+            'date_end': fields.Datetime.to_string(datetime.today() +
+                                                  timedelta(hours=4)),
+        }
+        task = self.Tasks.create(check_task_state_vals)
+        self.assertEqual(task.name, 'Sample Task for checking task state')
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py]: No need to check this

File: project_event/tests/test_project_event_task.py [hunk 10]
```diff
@@ -527,3 +527,37 @@ def test_220_get_parent_project_id(self):
...
+        task.action_option()
+        self.assertEqual(task.task_state, 'option')
+        check_task_state = task.check_task_state(task.task_state)
+        self.assertEqual(check_task_state, True)
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py]: simply check these 2 cases:
self.assertTrue(self.task1.check_task_state('draft'))
self.assertFalse(self.task1.check_task_state('......any state that is not in the test domain'))

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

**Review by rimbendhaou — APPROVED** [review]

**Review by eilst — COMMENTED** [review]
LGTM

**Pull request merged**